npm_and_yarn/lib/dependabot/npm_and_yarn/package_manager.rb

```diff
@@ -66,6 +265,17 @@ def setup(name)
 
       private
 
+      sig { params(name: String).returns(Ecosystem::VersionManager) }
+      def package_manager_by_name(name)
+        package_manager_class = PACKAGE_MANAGER_CLASSES[name]
+
+        package_manager_class ||= PACKAGE_MANAGER_CLASSES[DEFAULT_PACKAGE_MANAGER]
+
+        version = Helpers.send(:"#{name}_version_numeric", @lockfiles[name.to_sym])
+
+        package_manager_class.new(version.to_s)
+      end
```

Comment:
We want to initiate package manager by it is name and for default we are using `npm` if we are unable to detect package manager name from dependency_files.